### PR TITLE
Reflection in script fixes

### PR DESCRIPTION
**Pull request by Sajdkick**
Moved the filling of the property map to the script component, also made sure that the building of the script was successful before fetching the properties

### CODE REVIEW

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 6 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [ScriptManager.cpp#L668](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/df50db9d9caf4dfc4daadc0179fe1de5596ec314/src/Engine/Manager/ScriptManager.cpp#L668): Memory leak: value [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3Amemleak)
1. ![MAJOR][MAJOR] [ScriptManager.cpp#L672](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/df50db9d9caf4dfc4daadc0179fe1de5596ec314/src/Engine/Manager/ScriptManager.cpp#L672): Memory leak: value [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3Amemleak)
1. ![MAJOR][MAJOR] [ScriptManager.cpp#L676](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/df50db9d9caf4dfc4daadc0179fe1de5596ec314/src/Engine/Manager/ScriptManager.cpp#L676): Memory leak: value [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3Amemleak)
1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [ScriptManager.cpp#L690](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/df50db9d9caf4dfc4daadc0179fe1de5596ec314/src/Engine/Manager/ScriptManager.cpp#L690): The function 'GetScripts' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![INFO][INFO] [ScriptManager.cpp#L466](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/df50db9d9caf4dfc4daadc0179fe1de5596ec314/src/Engine/Manager/ScriptManager.cpp#L466): comma should be followed by whitespace [rule](http://localhost:9000/coding_rules#rule_key=vera%2B%2B%3AT009)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/Engine/Component/Script.cpp [hunk 1]
```diff
@@ -50,6 +50,51 @@ Json::Value Script::Save() const {
...
+            }
+
+        }
+
+        if (typeId == asTYPEID_INT32) {
+            int* mapValue = new int();
+            *mapValue = *(int*)varPointer;
+            propertyMap[instance->GetPropertyName(n)] = std::pair<int, void*>(typeId, mapValue);
+        }
+        else if (typeId == asTYPEID_FLOAT) {
```
Comment (Chainsawkitten) [inline src/Engine/Component/Script.cpp]: Bracket and else should be on same line

File: src/Engine/Component/Script.cpp [hunk 2]
```diff
@@ -50,6 +50,51 @@ Json::Value Script::Save() const {
...
+
+    for (int n = 0; n < propertyCount; n++) {
+
+        int typeId = instance->GetPropertyTypeId(n);
+        void *varPointer = instance->GetAddressOfProperty(n);
+
+        auto it = propertyMap.find(instance->GetPropertyName(n));
+        if (it != propertyMap.end()) {
+
+            if (propertyMap[instance->GetPropertyName(n)].first == typeId) {
```
Comment (Chainsawkitten) [inline src/Engine/Component/Script.cpp]: No brackets for single line if.

File: src/Engine/Manager/ScriptManager.cpp [hunk 3]
```diff
@@ -434,48 +449,16 @@ void ScriptManager::BuildAllScripts() {
...
-            std::string *str = (std::string*)varPointer;
-            if (str) {
-
-                std::string* mapValue = new std::string();
-                *mapValue = *(std::string*)varPointer;
-                script->propertyMap[script->instance->GetPropertyName(n)] = std::pair<int, void*>(typeId, mapValue);
-
-            }
+    }
+    else {
```
Comment (Chainsawkitten) [inline src/Engine/Manager/ScriptManager.cpp]: Bracket and else should be on same line.

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 6 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [ScriptManager.cpp#L668](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/715d3f211a6043702a1f8e2b32fe890692069212/src/Engine/Manager/ScriptManager.cpp#L668): Memory leak: value [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3Amemleak)
1. ![MAJOR][MAJOR] [ScriptManager.cpp#L672](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/715d3f211a6043702a1f8e2b32fe890692069212/src/Engine/Manager/ScriptManager.cpp#L672): Memory leak: value [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3Amemleak)
1. ![MAJOR][MAJOR] [ScriptManager.cpp#L676](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/715d3f211a6043702a1f8e2b32fe890692069212/src/Engine/Manager/ScriptManager.cpp#L676): Memory leak: value [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3Amemleak)
1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [ScriptManager.cpp#L690](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/715d3f211a6043702a1f8e2b32fe890692069212/src/Engine/Manager/ScriptManager.cpp#L690): The function 'GetScripts' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![INFO][INFO] [ScriptManager.cpp#L466](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/715d3f211a6043702a1f8e2b32fe890692069212/src/Engine/Manager/ScriptManager.cpp#L466): comma should be followed by whitespace [rule](http://localhost:9000/coding_rules#rule_key=vera%2B%2B%3AT009)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 6 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [ScriptManager.cpp#L668](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/ec435992664f0ac8cc2505eb713a56d8b3f74088/src/Engine/Manager/ScriptManager.cpp#L668): Memory leak: value [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3Amemleak)
1. ![MAJOR][MAJOR] [ScriptManager.cpp#L672](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/ec435992664f0ac8cc2505eb713a56d8b3f74088/src/Engine/Manager/ScriptManager.cpp#L672): Memory leak: value [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3Amemleak)
1. ![MAJOR][MAJOR] [ScriptManager.cpp#L676](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/ec435992664f0ac8cc2505eb713a56d8b3f74088/src/Engine/Manager/ScriptManager.cpp#L676): Memory leak: value [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3Amemleak)
1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [ScriptManager.cpp#L690](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/ec435992664f0ac8cc2505eb713a56d8b3f74088/src/Engine/Manager/ScriptManager.cpp#L690): The function 'GetScripts' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![INFO][INFO] [ScriptManager.cpp#L466](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/ec435992664f0ac8cc2505eb713a56d8b3f74088/src/Engine/Manager/ScriptManager.cpp#L466): comma should be followed by whitespace [rule](http://localhost:9000/coding_rules#rule_key=vera%2B%2B%3AT009)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**